civiform/civiform
# Adding prefix to error toast messages, localized when applicant-facing, for accessibility

### Description

- Added "Error: " as a prefix on toast messages, as an accessibility enhancement.  For toast messages which were already being localized, the prefix will also be localized.  For other admin-facing toast messages, the prefix is added in English.
-  Turned ToastMessage.java into more of a factory class for consistency in how we create toasts and so that new error toast messages will include the prefix.
- Note that by moving this prefix logic server-side, any client-side-only toast messages will not include the prefix.  So far, I have only seen one instance of a client-side-only toast and that was during admin predicates editing.  I will create a tech debt item to address that edge-case.  

## Release notes

Added "Error: " as a prefix on toast messages, as an accessibility enhancement.

### Checklist

#### General

Read the full guidelines for PRs [here](https://docs.civiform.us/contributor-guide/developer-guide/technical-contribution-guide#guidelines-for-pull-requests)


#### User visible changes

- [x] Followed steps to [internationalize new strings](https://docs.civiform.us/contributor-guide/developer-guide/internationalization-i18n#internationalization-for-application-strings)

### Instructions for manual testing

1. As a CF admin, set a visibility condition on a program block
2. Publish the program
3. Log out
4. Open the application for the program
5. Fill out the question with an answer that doesn't display the next block. Save and go to the review page.
6. Clone the tab
7. In the new tab, change the answer to the question to an answer that would display the next block and save.
8. Go back to the original tab and hit submit
9. An error toast should show saying the application has changed.  It should begin with "Error: " and this prefix should be localized.

### Issue(s) this completes

Fixes [#3286](https://github.com/civiform/civiform/issues/3286);
